test/optional_test.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Google LLC
```

Comment:
Similar to common.h / common.c, This could be in its own PR it you wished. Since the optional tests are not blocking other PRs I'll leave it up to you.